python/xoscar/collective/process_group.py

```diff
@@ -321,3 +329,338 @@ def broadcast(
             root,
             tag,
         )
+
+
+class ProcessGroupNCCL(ProcessGroup):
+    def __init__(
+        self,
+        ip: str,
+        rank: int,
+        device_id: int,
```

Comment:
I think this ``device_id`` should not be a required parameter. By default, ``device_id`` should be equal to ``rank``.